src/lib/autodiscovery/autodiscovery.py

```diff
@@ -131,7 +215,28 @@ async def get_metric_descriptors(
             fetch_reslut.project_id
         )
 
-    return metric_per_project
+    return metric_per_project, resource_dimensions
+
+
+async def get_existing_metrics(
+    autodiscovery_resources: Dict[str, AutodiscoveryResourceLinking],
+) -> Dict[str, Set[str]]:
+    resources_to_metrics = {
+        resource_name: set() for resource_name in autodiscovery_resources.keys()
+    }
+
+    for resource_name, service_linking in autodiscovery_resources.items():
```

Comment:
At some moment I lost inside the level of the nested loops

Reply:
True. It's worth to check if we can reduce all the nesting (other places too). Maybe it could be simplified.